assets/app/features/upgrade/controllers/upgrade-database-configuration.controller.spec.js

```diff
@@ -42,7 +43,14 @@ describe('Upgrade Flow - Create Connect Database Controller', function () {
     beforeEach(function() {
         //Setup the module and dependencies to be used.
         bard.appModule('crowbarApp.upgrade');
-        bard.inject('$controller', '$q', '$httpBackend', '$rootScope', 'upgradeFactory');
+        bard.inject(
+            '$controller', '$q', '$httpBackend', '$rootScope',
+            'upgradeFactory', 'upgradeStatusFactory', 'upgradeStepsFactory',
+            'UPGRADE_STEPS'
+        );
+
+        spyOn(upgradeStatusFactory, 'syncStatusFlags');
+        spyOn(upgradeStepsFactory, 'setCurrentStepCompleted');
```

Comment:
`upgradeStepsFactory` is referred to in line 75.

PS. this was your idea to mock this function and use it as argument to `toHaveBeenCalledWith()` :)